daffodil-sapi/src/main/scala/org/apache/daffodil/sapi/Daffodil.scala

```diff
@@ -105,6 +106,10 @@ object ValidationMode extends Enumeration {
   val Off = Value(10)
   val Limited = Value(20)
   val Full = Value(30)
+
+  case class Custom(v: Validator) extends ValidationMode {
```

Comment:
After this Validator stuff, we have to figure out how to get rid of the redundancy this API/JAPI/SAPI stuff creates. There is way too much stuff to keep in sync, and way too much duplication, and it gets in the way of easy-extensibility terribly.